test/integration/src/test/java/com/liferay/faces/portal/test/integration/demo/PrimeFacesUsersPortletTester.java

```diff
@@ -48,506 +52,621 @@
-		// permitted.
-		browserStateAsserter.assertElementDisplayed(
-			"//input[contains(@id,':emailAddress')]/../span[@class='portlet-msg-error' and text()='Invalid Email Address']");
+	@Test
+	public void runPrimeFacesUsersPortletTest_B_UsersSort() {
 
-		// Clear the "First Name", "Last Name", and "Email Address" fields and click the "Submit" button.
-		browserDriver.clearElement(FIRST_NAME_FIELD_XPATH);
-		browserDriver.clearElement(LAST_NAME_FIELD_XPATH);
-		browserDriver.clearElement(EMAIL_ADDRESS_FIELD_XPATH);
-		browserDriver.clickElement(submitButtonXpath);
-
-		// Verify that the error message "Value is required" is displayed for "First Name", "Last Name", and "Email
-		// Address" since they are all required fields.
-		browserStateAsserter.assertElementDisplayed(
-			"//input[contains(@id,':firstName')]/../span[@class='portlet-msg-error' and text()='Value is required']");
-		browserStateAsserter.assertElementDisplayed(
-			"//input[contains(@id,':lastName')]/../span[@class='portlet-msg-error' and text()='Value is required']");
-		browserStateAsserter.assertElementDisplayed(
-			"//input[contains(@id,':emailAddress')]/../span[@class='portlet-msg-error' and text()='Value is required']");
-
-		// Click on the "Cancel" button in order to return to the list of users.
-		browserDriver.clickElement("//button[contains(@id, ':pushButtonCancel') and @type='submit']");
-		browserDriver.waitForElementDisplayed(FIRST_USER_SCREEN_NAME_XPATH);
-
-		// TODO: It should not be necessary to re-enter "john.adams" into the Screen Name filter but if not done, then
-		// the test has subsequent assertion failures.
-		filterByScreenNameAndVerifySingleUserDisplayed(browserDriver, browserStateAsserter, "john.adams");
-
-		// Clear the Screen Name filter and verify that the list contains more than one user.
-		browserDriver.clearElement(SCREEN_NAME_COLUMN_FILTER_XPATH);
-		browserStateAsserter.assertElementDisplayed(SECOND_USER_SCREEN_NAME_CELL_XPATH);
+		BrowserDriver browserDriver = getBrowserDriver();
 
-		// Verify that when "Adams" is entered into the last name filter that only users with the last name "Adams"
-		// are present in the list.
-		String lastNameColumnFilterXpath = "//input[contains(@id,'users:lastName:filter')]";
-		Actions filterActions = browserDriver.createActions(lastNameColumnFilterXpath);
-		WebElement lastNameColumnFilterElement = browserDriver.findElementByXpath(lastNameColumnFilterXpath);
-		filterActions.sendKeys(lastNameColumnFilterElement, "Adams");
-		browserDriver.performAndWaitForRerender(filterActions.build(), "(//span[contains(@id,':firstNameCell')])[1]");
+		// 1. Take note of each screen name in the first column of the table, clicking the *Next* button until all
+		// the screen names have been noted.
+		List<String> screenNames = extractColumnValuesFromDataTable(browserDriver, "screenName");
+		logList(screenNames, "Screen names extracted from the UI:");
```

Comment:
@stiemannkj1 FWIW, the reason why I we tend to avoid logger utility methods (to avoid duplicate code) is because the line numbers in the logger lose the context of the line number. But in the case of `logList` it seems fine, since the message is different each time it is called.